x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/60_semantic_text_inference_update.yml

```diff
@@ -0,0 +1,612 @@
+  - match:  { _source.dense_field.inference.chunks.0.embeddings: $dense_field_embedding }
+  - match:  { _source.non_inference_field: "another non inference test" }
+
+---
+"Updating semantic_text fields recalculates embeddings":
+  - do:
+      index:
+        index: test-index
+        id: doc_1
+        body:
+          sparse_field: "inference test"
+          dense_field: "another inference test"
+          non_inference_field: "non inference test"
+
+  - do:
+      get:
+        index: test-index
+        id: doc_1
+
+  - match:  { _source.sparse_field.text: "inference test" }
+  - match:  { _source.sparse_field.inference.chunks.0.text: "inference test" }
+  - match:  { _source.dense_field.text: "another inference test" }
+  - match:  { _source.dense_field.inference.chunks.0.text: "another inference test" }
+  - match:  { _source.non_inference_field: "non inference test" }
+
+  - do:
+      bulk:
+        index: test-index
+        body:
+          - '{"update": {"_id": "doc_1"}}'
+          - '{"doc":{"sparse_field": "I am a test", "dense_field": "I am a teapot"}}'
+
+  - do:
+      get:
+        index: test-index
+        id: doc_1
+
+  - match:  { _source.sparse_field.text: "I am a test" }
+  - match:  { _source.sparse_field.inference.chunks.0.text: "I am a test" }
+  - match:  { _source.dense_field.text: "I am a teapot" }
+  - match:  { _source.dense_field.inference.chunks.0.text: "I am a teapot" }
+  - match:  { _source.non_inference_field: "non inference test" }
+
+  - do:
+      update:
+        index: test-index
+        id: doc_1
+        body:
+          doc:
+            sparse_field: "updated inference test"
+            dense_field: "another updated inference test"
+
+  - do:
+      get:
+        index: test-index
+        id: doc_1
+
+  - match:  { _source.sparse_field.text: "updated inference test" }
+  - match:  { _source.sparse_field.inference.chunks.0.text: "updated inference test" }
+  - match:  { _source.dense_field.text: "another updated inference test" }
+  - match:  { _source.dense_field.inference.chunks.0.text: "another updated inference test" }
+  - match:  { _source.non_inference_field: "non inference test" }
+
+  - do:
+      bulk:
+        index: test-index
+        body:
+          - '{"update": {"_id": "doc_1"}}'
+          - '{"doc":{"sparse_field": "bulk inference test", "dense_field": "bulk updated inference test"}}'
+
+  - do:
+      get:
+        index: test-index
+        id: doc_1
+
+  - match:  { _source.sparse_field.text: "bulk inference test" }
+  - match:  { _source.sparse_field.inference.chunks.0.text: "bulk inference test" }
+  - match:  { _source.dense_field.text: "bulk updated inference test" }
+  - match:  { _source.dense_field.inference.chunks.0.text: "bulk updated inference test" }
+  - match:  { _source.non_inference_field: "non inference test" }
+
+---
+"Update logic handles source fields in object fields":
```

Comment:
I also wanted to test source fields with dots in the name, but I think I found a deeper bug that we would need to address first.

If you create a mapping like this:

```json
PUT sample-index
{
  "mappings": {
    "properties": {
      "inference_field": {
        "type": "semantic_text",
        "inference_id": "my-inference-endpoint"
      },
      "source": {
        "type": "object",
        "subobjects": false
      },
      "source.foo.bar": {
        "type": "text",
        "copy_to": "inference_field"
      }
    }
  }
}
```

You'll end up with the field `foo.bar` in the `source` object. However, `foo.bar` is _not_ registered as a source for `inference_field`. I traced this behavior all the way back to the [FieldTypeLookup constructor](https://github.com/elastic/elasticsearch/blob/25e35510ab0ea75e12be6dc72a47c2e06e08fccb/server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java#L46); it has nothing to do with `semantic_text`.

If this is not the intended behavior, I'll file a bug report and we can fix the issue in a separate PR.